packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -954,8 +954,41 @@ actual object RealmInterop {
         )
     }
 
-    actual fun realm_sync_client_config_new(): RealmSyncClientConfigurationPointer {
-        return LongPointerWrapper(realmc.realm_sync_client_config_new())
+    actual fun realm_sync_client_config_new(appId: String): RealmSyncClientConfigurationPointer {
+        val ptr: RealmSyncClientConfigurationPointer = LongPointerWrapper(realmc.realm_sync_client_config_new())
+        realmc.realm_sync_client_config_set_default_binding_thread_observer(
```

Comment:
Hmm. I am a little against having this kind of logic inside the `realm_sync_client_config_new` as `cinterop` usually is isolated to dispatch individual calls to the C-API. But I guess we are also trying to make RealmInterop platform agnostic 🤔 There are other cases where we have pushed common code into RealmInterop which sort of goes the other direction requiring us to keep the platform implementations in sync manually, i.e.
- https://github.com/realm/realm-kotlin/blob/main/packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt#L1330
- https://github.com/realm/realm-kotlin/blob/main/packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt#L1152

I guess the ideal solution would be to have a platform agnostic API at the `cinterop` layer with ability to include implementation of common things. I guess it is a bit overkill to do as part of this PR and fits better into a task like #792, but I think it is still valid to have in mind going forward and should maybe push for prioritization of #792.

Reply:
I think I would feel it would be safer, as these kind of things would be less hidden. Imagining introducing a new platform in `cinterop` - then it is just a matter of binding the C-API and now replicating other logic into various methods. But it would then be an issue knowing that `realm_sync_client_config_set_default_binding_thread_observer` is a no-op on Darwin and not trust it to take care of something that it doesn't do. But we could also just do the full implementation on Darwin if it made sense for logging thread creation/teardown 🤔 I guess a single method registration call and a callback wouldn't matter that much on this code path anyway.

Reply:
I reworked it so there is also now a thread observer on Darwin. We don't need it for cleaning up threads, but at least handling errors on the Sync thread will be done in the same now.